src/Logger.cc

```diff
@@ -415,8 +415,8 @@ QString Logger::userName()
 
     if ( pw )
         return pw->pw_name;
-    else
-        return QString::number( getuid() );
+
```

Comment:
Same: This pretends that the error case (not having a password file entry) were the norm. No, it's not; it's an exceptional case. It's not really an error since any system might have user IDs in use that are not in `/etc/passwd`, but it's still a very rare case.

And again, the compiler does complain if there is any chance that none of those branches are executed.